jboss-eap-config-jgroups/added/launch/jgroups.sh

```diff
@@ -0,0 +1,170 @@
+validate_keystore_legacy() {
+    declare jg_encrypt_secret="$1" jg_encrypt_name="$2" jg_encrypt_password="$3" jg_encrypt_keystore="$4" jg_encrypt_keystore_dir="$5"
+    if [ -n "${jg_encrypt_secret}"   -a \
+      -n "${jg_encrypt_name}"         -a \
+      -n "${jg_encrypt_password}"     -a \
+      -n "${jg_encrypt_keystore_dir}" -a \
+      -n "${jg_encrypt_keystore}" ]; then
+        echo "valid"
+      elif [ -n "${jg_encrypt_secret}" ]; then
+        echo "partial"
+      else
+        echo "missing"
+      fi
+}
+
+configure_jgroups_encryption() {
+ local jgroups_encrypt_protocol="${JGROUPS_ENCRYPT_PROTOCOL:=SYM_ENCRYPT}"
+ local jgroups_encrypt=""
+ local key_store=""
+ case "${jgroups_encrypt_protocol}" in
+  "SYM_ENCRYPT")
+    log_info "Configuring JGroups cluster traffic encryption protocol to SYM_ENCRYPT."
+    local jgroups_unencrypted_message="Detected <STATE> JGroups encryption configuration, the communication within the cluster WILL NOT be encrypted."
+    local keystore_warning_message=""
+    local has_elytron_tls_marker=$(has_elytron_tls "${CONFIG_FILE}")
+    local keystore_validation_state="";
+    if [ "${has_elytron_tls_marker}" = "true" ]; then
+        keystore_validation_state=$(validate_keystore "${JGROUPS_ENCRYPT_SECRET}" "${JGROUPS_ENCRYPT_NAME}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_KEYSTORE}")
+    else
+        keystore_validation_state=$(validate_keystore_legacy "${JGROUPS_ENCRYPT_SECRET}" "${JGROUPS_ENCRYPT_NAME}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_KEYSTORE_DIR}")
+    fi
+
+    if [ "${keystore_validation_state}" = "valid" ]; then
+        # first add the elytron key-store:
+        if [ "${has_elytron_tls_marker}" = "true" ]; then
+            key_store=$(create_elytron_keystore "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_KEYSTORE_TYPE}" "${JGROUPS_ENCRYPT_KEYSTORE_DIR}")
+            jgroups_encrypt=$(create_jgroups_elytron_encrypt_sym "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_NAME}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_ENTIRE_MESSAGE:-true}")
+        else
+            # compatibility with old marker, only used if new marker is not present
+            jgroups_encrypt=$(create_jgroups_elytron_legacy "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_NAME}" "${JGROUPS_ENCRYPT_KEYSTORE_DIR}")
+        fi
+    elif [ "${keystore_validation_state}" = "partial" ]; then
+        keystore_warning_message="${jgroups_unencrypted_message//<STATE>/partial}"
+    else
+         keystore_warning_message="${jgroups_unencrypted_message//<STATE>/missing}"
+    fi
+
+    if [ -n "${keystore_warning_message}" ]; then
+        log_warning "${keystore_warning_message}"
+    fi
+
+    # add elytron block if we need it
+    if [ "${has_elytron_tls_marker}" = "true" ]; then
+        insert_elytron_tls
+    fi
```

Comment:
I think similar to what you did in elytron.sh, this insert_elytron_tls at L140 should go in the if block at L121. IOW don't add the elytron xml structure unless you plan to replace the elements.

Reply:
+1 yep, I missed this when I was fixing the other one.